### PR TITLE
Adds test flows for OCP 4.5

### LINKED ISSUE #265: Secret provider support ocp
## Add and remove support of OCP

- [ ] Remove support of OCP 3.9 and 3.10 in tests and read me

- [ ] Add support of OCP 4.5 tests and read me or create a page with delivery plan (depend on the EE)

- [ ] Make sure Secret provider system requirements are part of docs like: 
  - [ ] https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/scl_SystemReq.htm
  - [ ] https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/Integrations/k8s-ocp/cjr-secrets-provider-lp.htm#Systemrequirements

- Note that these requirements are hard to find in our documentation, they are scattered in too many places. We should have a solid reference with full coverage from the main [system requirements](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/Deployment/platforms/dap-sysreqs-server.htm?tocpath=Setup%7CDAP%20Requirements%7CDAP%20System%20Requirements%7C_____1) section

### LINKED ISSUE #275: Remove support of OCP 3.09 &3.10 from documents
Remove support of OCP 3.09 &3.10 from documents
-[ ] Update SP Read me

### PR DESCRIPTION
**Pull request by izgeri**
### What does this PR do?
- Removes test flows for OCP versions under 3.11
- Updates documented supported versions of OCP in README and adds deprecation
  notice in CHANGELOG

Note: this code just adds the test flows, but I will be surprised if it passes as-is. It's likely that some other small changes will need to be made to get this working.

### What ticket does this PR close?
Resolves #265 
Resolves #275 

#### Change log
- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a CHANGELOG update

#### Test coverage
- [x] This PR includes new unit and integration tests to go with the code changes, or
- [ ] The changes in this PR do not require tests

#### Documentation
- [x] Docs (e.g. `README`s) were updated in this PR, and/or there is a follow-on issue to update docs, or
- [ ] This PR does not require updating any documentation

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -99,10 +99,6 @@ pipeline {
                 // tasks["Openshift v3.11, oss"] = {
                 //  sh "./bin/start --docker --oss --oc311"
                 // }
```
Comment (orenbm) [inline Jenkinsfile:101]: @eladkug @InbalZilberman do we have a plan to run tests of Conjur OSS with OCP? if not, can we remove this to enhance code maintainability?

**Comment by orenbm** [comment]
thanks for this effort @izgeri :)

File: Jenkinsfile (hunk 1 above)
Comment (eladkug) [reply Jenkinsfile:101]: we can remove this. We dont support it (document updated as well in the past)
`## Supported Platforms
- GKE

- Openshift 3.9, 3.10, and 3.11 _*(DAP only)*_`

Comment (orenbm) [reply Jenkinsfile:101]: thanks @eladkug .

@izgeri can you please remove these commented out tasks while you're at it?

File: deploy/utils.sh [hunk 2]
```diff
@@ -102,6 +102,8 @@ runDockerCommand() {
     -e OPENSHIFT_PASSWORD \
     -e DOCKER_REGISTRY_PATH \
     -e DOCKER_REGISTRY_URL \
+    -e PULL_DOCKER_REGISTRY_PATH \
+    -e PULL_DOCKER_REGISTRY_URL \
```
Comment (orenbm) [inline deploy/utils.sh:106]: @izgeri can you elaborate on the difference between these 2?

**Review by orenbm — APPROVED** [review]
LGTM, thanks for getting this!

File: deploy/utils.sh (hunk 2 above)
Comment (izgeri) [reply deploy/utils.sh:106]: I'm following a pattern we used in k8s-demo when we added 4.3/4.5 support [PR](https://github.com/conjurdemos/kubernetes-conjur-demo/pull/124/files) - maybe @doodlesbykumbi can elaborate?

Comment (izgeri) [reply deploy/utils.sh:106]: Looking at the table in the PR I linked:
- `DOCKER_REGISTRY_URL` is the Docker registry URL
- `DOCKER_REGISTRY_PATH` is `{DOCKER_REGISTRY_URL}/{path}` where `{path}` is any platform specific organization path
- `PULL_DOCKER_REGISTRY_URL` is the Docker registry URL you use for _pulling_ images - in OCP 4.x, we have to use an internal registry URL for pulling images or the image pull fails - so there's a different URL for pushing (which is done from your local machine or an executor, e.g. external to the cluster) vs pulling (internal to the cluster)
- `PULL_DOCKER_REGISTRY_PATH` is `PULL_DOCKER_REGISTRY_URL/{path}` where, as above, `{path}` is any platform specific organization path

**Review by diverdane — COMMENTED** [review]
This LGTM! Added one comment about possibly adding a handy pointer to the env variable meanings in kubernetes-conjur-demo, but not a big deal.

**Comment by orenbm** [comment]
> Just a thought, maybe it would be helpful to have a comment at the top of this function saying to look in the kubernetes-conjur-demo for meaning/usage of environment variables?

@diverdane can you elaborate on this? i couldn't find any description on this in `kubernetes-conjur-demo`

**Comment by diverdane** [comment]
> > Just a thought, maybe it would be helpful to have a comment at the top of this function saying to look in the kubernetes-conjur-demo for meaning/usage of environment variables?
> 
> @diverdane can you elaborate on this? i couldn't find any description on this in `kubernetes-conjur-demo`

I meant adding a "cross-repo" pointer to this table: https://github.com/conjurdemos/kubernetes-conjur-demo#usage-instructions

File: deploy/platform_login.sh [hunk 3]
```diff
@@ -16,11 +16,9 @@ main() {
       -u oauth2accesstoken \
       -p "$(gcloud auth print-access-token)"
   elif [[ "$PLATFORM" = "openshift" ]]; then
-    # sensible default for OPENSHIFT_URL port
-    if [[ -n "${OPENSHIFT_URL}" ]] && [[ "${OPENSHIFT_URL}" != *: ]]; then
-     OPENSHIFT_URL="${OPENSHIFT_URL}:8443"
+    if [[ -n "${OPENSHIFT_URL}" ]] && [[ ! "${OPENSHIFT_URL}" =~ :[[:digit:]] ]]; then
```
Comment (sigalsax) [inline deploy/platform_login.sh:19]: would be helpful to comment what is being achieved here

Comment (orenbm) [reply deploy/platform_login.sh:21]: done, thanks

**Review by sigalsax — APPROVED** [review]
LGTM! ✨

**Comment by izgeri** [comment]
@orenbm do you think this will be merged soon? I'd like to get the change announcing the new OCP versions into the next suite release, which means merging this PR and tagging the project (#287). I am looking to wrap that up this week, if I can.

**Pull request merged**